Configuration for test-kitchen

geewiz opened the pull request:
This configuration file defines defaults for test-kitchen projects:
- Use Docker driver
- Pre-install Chef 12 and Busser packages for faster spin-up

@freistil/ops This configuration is highly recommended!

Commit:
Configuration for test-kitchen

This configuration file defines defaults for test-kitchen projects:

* Use Docker driver
* Pre-install Chef 12 and Busser packages for faster spin-up

muhh commented on kitchen/config.yml:
```diff
@@ -0,0 +1,12 @@
+driver_plugin: docker
+driver_config:
+  provision_command:
+    - curl -L https://www.opscode.com/chef/install.sh | bash -s -- -v 12.0.3
```
This installs an outdated chef version. And it pins the version. I'd prefer to just install the latest as default.

geewiz commented:
Pinning the version ensures that Docker only has to install it once and then use its cache for every new container. In think that in this context, test performance is more important than running the bleeding edge (of which I don't see the advantage).

muhh commented:
No need for bleeding edge. But it should at least be the current version we have rolled out everywhere.
No need to for keeping using 16 month old buggy chef-clients in development.

geewiz commented:
Ok. I'm testing with 12.9.41 and will push an update as soon as it's finished.

geewiz commented:
At least with the latest version, Busser (integration tests) throws an exception. That's why I added the version constraint (that I found on the web) in the first place; when that worked, I left it at the old version. I'll try and find out up to which Chef version Busser doesn't break.

Commit:
test-kitchen: Use Chef 12.6.0

Since 12.0 is too old and 12.9 breaks Busser, I've changed the version
to the one we're currently using in production.

geewiz commented:
It works with 12.6.0, the version we use in production. So I think it makes sense using that by default.
